generateIndices.py

```diff
@@ -0,0 +1,81 @@
+#reads in files from features.npz
+# output training and testing indices
+import sys
+import re
+import math
+import numpy as np
+
+def generateIndex():
+
+    feature_file = sys.argv[1]
+
+    features_files = np.load(feature_file)
+    filenames = features_files['filenames']
+
+    #master list to hold information
+    master = []
+    #number of files in each pile
+    numberOfFiles_0=0
+    numberOfFiles_1=0
+    numberOfFiles_2=0
+    numberOfFiles_3=0
+
+
+    pic_pattern = re.compile("^(\d)_(\d+).jpg$")
+
+    for i in range(filenames.shape[0]):
+        wholeFileName = filenames[i].split("/")[1]
+
+        m = pic_pattern.match(wholeFileName)
+        if m:
+            # label, number within group, index in the feature vector/matrix
+            master.append((int(m.group(1)), int(m.group(2)), i))
```

Comment:
Instead of referencing ```m.group(1)``` and converting it each time you can do
```
label, within_group_number = int(m.group(1)), int(m.group(2))
```
at the top and use the named variables everywhere.